### DISCUSSION TITLE
Composer require conflict

### DISCUSSION THREAD

**Discussion by simbiant** [spatie/laravel-ray#192] [Q&A]
Hi, who can tell me what's wrong

 Problem 1
    - spatie/laravel-ray[1.17.0, ..., 1.17.4] require illuminate/contracts ^7.20|^8.0 -> found illuminate/contracts[v7.20.0, ..., 7.x-dev, v8.0.0, ..., 8.x-dev] but these were not loaded, likely because it conflicts with another require.
    - Root composer.json requires spatie/laravel-ray ^1.17 -> satisfiable by spatie/laravel-ray[1.17.0, ..., 1.17.4].
    
    
    `{
    "name": "laravel/laravel",
    "description": "The Laravel Framework.",
    "keywords": ["framework", "laravel"],
    "license": "MIT",
    "type": "project",
    "require": {
        "php": "~7.4",
        "ext-bcmath": "*",
        "ext-gd": "*",
        "ext-json": "*",
        "ext-zip": "*",
        "doctrine/dbal": "^2.7",
        "dompdf/dompdf": "^0.8.2",
        "fideloper/proxy": "^4.0",
        "firebase/php-jwt": "^5.0",
        "google/apiclient": "^2.2",
        "google/cloud-pubsub": "v1.24",
        "laravel/framework": "*",
        "laravel/helpers": "^1.0",
        "lcobucci/jwt": "3.3.3",
        "laravel/passport": "7.5.1",
        "laravel/tinker": "^1.0",
        "laravelcollective/html": "5.8.*",
        "maatwebsite/excel": "^3.1",
        "maddhatter/laravel-fullcalendar": "^1.3",
        "nmred/kafka-php": "v0.2.0.8",
        "phpmailer/phpmailer": "^6.1",
        "phpoffice/phpspreadsheet": "^1.5",
        "psr/log": "v1.0.*",
        "sentry/sentry-laravel": "~1.0",
        "vinkla/hashids": "^5.0",
        "yajra/laravel-datatables-oracle": "~9.0",
    },
    "require-dev": {
        "barryvdh/laravel-debugbar": "^3.1",
        "barryvdh/laravel-ide-helper": "^2.6",
        "brianium/paratest": "^4.0",
        "filp/whoops": "^2.0",
        "fzaninotto/faker": "^1.4",
        "mockery/mockery": "^1.0",
        "nunomaduro/collision": "^2.0",
        "phpunit/phpunit": "^7.0",
        "slevomat/coding-standard": "^6.4",
        "squizlabs/php_codesniffer": "^3.5",
        "zircote/swagger-php": "^3.1"
    },
    "autoload": {
        "classmap": [
            "database/seeds",
            "packages/hurma/factories"
        ],
        "psr-4": {
            "App\\": "app/",
            "Hurma\\": "packages/hurma/src",
            "HurmaAPI\\": "packages/api/src",
            "Bots\\": "packages/bots/src"
        },
        "files": [
            "packages/functions/closures.php",
            "packages/functions/helpers.php"
        ]
    },
    "autoload-dev": {
        "classmap": [
            "packages/hurma/src"
        ],
        "psr-4": {
            "Hurma\\Tests\\": "packages/hurma/src/tests/"
        }
    },
    "extra": {
        "laravel": {
            "dont-discover": [
            ]
        }
    },
    "scripts": {
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "@php artisan key:generate"
        ],
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover"
        ],
        "swagger": "./vendor/bin/openapi --format json --output swagger.json ./packages/hurma/src/",
        "test": "./vendor/bin/phpunit -c phpunit.xml --process-isolation --verbose",
        "paratest": "./vendor/bin/paratest",
        "clear-logs": "echo '' > ./storage/logs/laravel.log",
        "code-check": [
            "./vendor/bin/phpcs -p --extensions=php --standard=phpcs.xml --colors --report-width=120 --parallel=8"
        ],
        "code-fix": [
            "./vendor/bin/phpcbf -p --extensions=php --standard=phpcs.xml --report=code --report-width=120 --parallel=8"
        ]
    },
    "config": {
        "preferred-install": "dist",
        "sort-packages": true,
        "optimize-autoloader": true
    },
    "minimum-stability": "dev",
    "prefer-stable": true
}
`

**Comment by sinnbeck** [comment]
Upgrade laravel/tinker to version 2. Version 1 requires laravel/contracts 6.0

https://github.com/laravel/tinker/blob/1.x/composer.json#L15
